freqtrade/exchange/exchange.py

```diff
@@ -1859,18 +1858,54 @@ def _get_funding_fees_from_exchange(self, pair: str, since: Union[datetime, int]
 
     @retrier
     def load_leverage_tiers(self) -> Dict[str, List[Dict]]:
-        if self.trading_mode == TradingMode.FUTURES and self.exchange_has('fetchLeverageTiers'):
-            try:
-                return self._api.fetch_leverage_tiers()
-            except ccxt.DDoSProtection as e:
-                raise DDosProtection(e) from e
-            except (ccxt.NetworkError, ccxt.ExchangeError) as e:
-                raise TemporaryError(
-                    f'Could not load leverage tiers due to {e.__class__.__name__}.'
-                    f'Message: {e}'
-                ) from e
-            except ccxt.BaseError as e:
-                raise OperationalException(e) from e
+        if self.trading_mode == TradingMode.FUTURES:
+            if self.exchange_has('fetchLeverageTiers'):
+                try:
+                    return self._api.fetch_leverage_tiers()
+                except ccxt.DDoSProtection as e:
+                    raise DDosProtection(e) from e
+                except (ccxt.NetworkError, ccxt.ExchangeError) as e:
+                    raise TemporaryError(
+                        f'Could not load leverage tiers due to {e.__class__.__name__}.'
+                        f'Message: {e}'
+                    ) from e
+                except ccxt.BaseError as e:
+                    raise OperationalException(e) from e
+            elif self.exchange_has('fetchMarketLeverageTiers'):
```

Comment:
I have to say - i do agree with flake8 - this method is too complex and should be split into 2 (it's really 2 methods ...).

I'd create 3 methods out fo this - one that selects the proper method and calls one of the 2 underlying methods for each ccxt method.

Reply:
I took a different approach that made more sense to me, but am willing to change it